src/rust/ide/lib/analytics/src/data.rs

```diff
@@ -0,0 +1,17 @@
+//! Provides data wrappers for our analytics api. This is intended to ensure we are conscious of
+//! whether we are sending public or private data. No private data should be logged at the moment.
+//!
+//! Note: this is meant to be a little bit un-ergonomic to ensure the data has been vetted by the
+//! API user and allow the reader of the code to see the intent behind the data.
+
+/// Wrapper struct for data that can be made public and has no privacy implications.
+#[derive(Clone,Copy,Debug)]
+pub struct Public<'a> {
+    pub data: &'a str
+}
```

Comment:
I don't think this is the best interface. Sometimes, you'd like to log a number, or just a "ping" that program is alive. Conversion of strings between Rust and JS is very pricy. There are special methods of conversion of other types, like nubers or empty tuple. So this should rather be 
```rust
pub struct Public<T> {
    pub data : T
}
``` 
instead.

Reply:
Moreover, I'm not sure this API prevents us really from doing something evil. When I see the line `remote_log(Public("node_added"))`, then `Public` tells me as much as `remote_log`. I mean, the name `Public` means something will be ... public, just as the word "remote". If we rename `Public` to `AnonymousData`, it would be much better IMO.